src/ngraph/runtime/cpu/op/quantized_avg_pool.cpp

```diff
@@ -0,0 +1,232 @@
+/*******************************************************************************
+* Copyright 2017-2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "quantized_avg_pool.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/util.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+op::QuantizedAvgPool::QuantizedAvgPool(const shared_ptr<Node>& arg,
+                                       const Shape& window_shape,
+                                       const Strides& window_movement_strides,
+                                       const Shape& padding_below,
+                                       const Shape& padding_above,
+                                       bool include_padding_in_avg_computation,
+                                       const shared_ptr<Node> min,
+                                       const shared_ptr<Node> max)
+    : Op("QuantizedAvgPool", check_single_output_args({arg, min, max}))
+    , m_window_shape(window_shape)
+    , m_window_movement_strides(window_movement_strides)
+    , m_padding_below(padding_below)
+    , m_padding_above(padding_above)
+    , m_include_padding_in_avg_computation(include_padding_in_avg_computation)
+{
+    constructor_validate_and_infer_types();
+
+    if (arg->get_element_type() != element::u8 && arg->get_element_type() != element::i8)
```

Comment:
why aren't these checks in `validate_and_infer`? I'm a bit confused by this split.